app/javascript/angularapps/openpicker/controllers/web_link_controller.js

```diff
@@ -0,0 +1,97 @@
+ WebLinkController = function($scope, $http, $timeout, DataAccessService, $sce, $filter, $location, Upload, OptionsService, LogService) {
+     var self = this;
+
+     self.scope = $scope;
+     self.timeout = $timeout;
+     self.http = $http;
+
+     self.sce = $sce;
+     self.location = $location;
+     self.Upload = Upload;
+     self.OptionsService = OptionsService;
+     self.logger = new LogService();
+
+
+     self.data_access_service = new DataAccessService();
+
+
+     self.setupScopeMethods();
+
+     self.initialize();
+
+
+
+ };
+
+ WebLinkController.prototype.setupScopeMethods = function() {
+     var self = this;
+     self.scope.fetch = function() {
+        
+         self.scope.showLoader = true;
```

Comment:
Done. Now the *view logic* and the *controller logic* is seperate in `web_link_controller.js`